src/canvas/SelectableCanvas.ts

```diff
@@ -854,41 +839,62 @@ export class SelectableCanvas<EventSpec extends CanvasEvents = CanvasEvents>
+   * and it remains unchanged from the viewer's perspective.
+   *
+   * @example
+   * const pointInCanvasPlane = sendPointToPlane(
+   *  this.getHTMLPointFromEvent(e),
+   *  undefined,
+   *  canvas.viewportTransform
+   * );
+   *
    */
-  restorePointerVpt(pointer: Point): Point {
-    return pointer.transform(invertTransform(this.viewportTransform));
+  getHTMLPointFromEvent(e: TPointerEvent) {
+    if (this._pointer) {
+      return this._pointer;
+    }
+    return this.getPointer(e, true);
   }
 
   /**
-   * Returns pointer coordinates relative to canvas.
-   * Can return coordinates with or without viewportTransform.
-   * ignoreVpt false gives back coordinates that represent
-   * the point clicked on canvas element.
-   * ignoreVpt true gives back coordinates after being processed
-   * by the viewportTransform ( sort of coordinates of what is displayed
-   * on the canvas where you are clicking.
-   * ignoreVpt true = HTMLElement coordinates relative to top,left
-   * ignoreVpt false, default = fabric space coordinates, the same used for shape position
-   * To interact with your shapes top and left you want to use ignoreVpt true
-   * most of the time, while ignoreVpt false will give you coordinates
-   * compatible with the object.oCoords system.
-   * of the time.
-   * @param {Event} e
-   * @param {Boolean} ignoreVpt
-   * @return {Point}
+   * @returns point existing in the canvas plane (the same plane as the plane {@link FabricObject#getCenterPoint} exists in),
+   * relative to instance's `(0, 0)`.
+   * This means that changes to the {@link viewportTransform} do not change the values of the point,
+   * however, from the viewer's perspective the point is changed.
+   *
+   * @example
+   * const pointInHTMLPlane = sendPointToPlane(
+   *  this.getCanvasPointFromEvent(e),
+   *  canvas.viewportTransform
+   * );
+   *
    */
-  getPointer(e: TPointerEvent, ignoreVpt = false): Point {
-    // return cached values if we are in the event processing chain
-    if (this._absolutePointer && !ignoreVpt) {
+  getCanvasPointFromEvent(e: TPointerEvent) {
+    if (this._absolutePointer) {
       return this._absolutePointer;
     }
-    if (this._pointer && ignoreVpt) {
-      return this._pointer;
-    }
+    return this.getPointer(e);
+  }
 
+  /**
+   * Returns pointer relative to canvas.
+   *
+   * **IMPORTANT**:
+   * This method is protected since v6 to protect you from misuse.
+   * Use {@link getHTMLPointFromEvent}, {@link getCanvasPointFromEvent} instead.
+   *
+   * @param {Event} e
+   * @param {Boolean} inHTMLPlane
+   * `true` returns a point that from the viewer's perspective remains untouched
+   * by {@link viewportTransform} changes\
+   * `false` returns a point in the canvas plane,
+   * the same plane as the plane {@link FabricObject#getCenterPoint} exists in
+   * @return {Point}
+   */
+  protected getPointer(e: TPointerEvent, inHTMLPlane = false): Point {
```

Comment:
please restore getPointer as normal method, we are not adding another migration headache to devs.

Reply:
you mean remove the protected keyword?

Reply:
I can do that.
As I see it it is a present to the dev.
TS will warn them about it. Then they can read the comment and decide if to change it or not. It is not a must. But it might help devs when debugging a hard bug.

Reply:
![image](https://github.com/fabricjs/fabric.js/assets/34343793/d339984f-cbb7-48a7-91c7-b217c25c74ad)

Reply:
Apart from saving the devs, having it protected saves fabric devs - I care about that a lot.
But it is not a must if you disapprove

Reply:
we can let it public and deprecated and later it will be protected with an internal name. we can't remove it now, i think i have hundreds of getPointer usage and so have others.

Reply:
Very well